src/lib/logstore/log_store.cpp

```diff
@@ -177,6 +177,7 @@ void HomeLogStore::on_log_found(logstore_seq_num_t seq_num, const logdev_key& ld
 
 void HomeLogStore::truncate(logstore_seq_num_t upto_lsn, bool in_memory_truncate_only) {
     if (upto_lsn < m_start_lsn) { return; }
+    flush();
```

Comment:
why do we need a flush here ?

Reply:
if the caller does not do a explicitly flush before truncating, then `m_tail_lsn` will not be updated and then truncating might not be able to truncate to the expected on.
add a `flush` here to make sure we do a flush before truncating. if flush is already scheduled by the called before truncating. then the flush here will do nothing and just return.